blocks/graph.py

```diff
@@ -354,3 +414,104 @@ def apply_noise(computation_graph, variables, level, seed=None):
         replace[variable] = (variable +
                              rng.normal(variable.shape, std=level))
     return computation_graph.replace(replace)
+
+
+def apply_dropout(computation_graph, variables, drop_prob=0.5, rng=None,
```

Comment:
For the input the probability is much more data-dependent I believe, and you might not want to drop out at all. I am just saying that 0.5 is a very reasonable default that works in 90% of the cases. Pylearn2, Caffe, Lasagne, Torch all have it.